### PR TITLE
[1.25] CVE april 26 patch

### PR DESCRIPTION
**Pull request by jbohanon**

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
         # https://github.com/envoyproxy/envoy/pull/26311
         #   ^ add dynamic metadata to MatchingData
-        #     https://github.com/solo-io/envoy-fork/commit/71359920812f13130e657c25913c8b174852daf4
-        #     https://github.com/ashishb-solo/envoy/tree/ashishb-solo/add-dynamic-metadata-to-matchingdata
-        commit = "fa3b44a537def7ab966e2192e2693be9bede0446",
+        #     https://github.com/envoyproxy/envoy/commit/1b66c3690422fb0d423411ebe70c777b9afbf2da
+        commit = "f89e0efddb8e4f8bad556608aa0e68fd5eae8d37",
         remote = "https://github.com/solo-io/envoy-fork",
+        patches = [
+            "@envoy_gloo//bazel/foreign_cc/filter-state-matching-input-25.yaml", # https://github.com/envoyproxy/envoy/pull/25856
```
Comment (ben-taussig-solo) [inline bazel/repository_locations.bzl:13]: Thanks for including the links to the PRs here!

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
         #   ^ add filter state matching input
         #     https://github.com/envoyproxy/envoy/commit/f52d559e0479824b9c964e4c028fa373bcb9b767
         # https://github.com/envoyproxy/envoy/pull/26311
         #   ^ add dynamic metadata to MatchingData
-        #     https://github.com/solo-io/envoy-fork/commit/71359920812f13130e657c25913c8b174852daf4
-        #     https://github.com/ashishb-solo/envoy/tree/ashishb-solo/add-dynamic-metadata-to-matchingdata
-        commit = "fa3b44a537def7ab966e2192e2693be9bede0446",
+        #     https://github.com/envoyproxy/envoy/commit/1b66c3690422fb0d423411ebe70c777b9afbf2da
+        commit = "f89e0efddb8e4f8bad556608aa0e68fd5eae8d37",
         remote = "https://github.com/solo-io/envoy-fork",
```
[the remark this answers is not in the document]
Comment (jbohanon) [reply bazel/repository_locations.bzl:11]: Nope, I just missed it.

File: bazel/repository_locations.bzl [hunk 3]
```diff
@@ -1,15 +1,18 @@
 REPOSITORY_LOCATIONS = dict(
     envoy = dict(
-        # envoy fork starting from 1.25.4 with cherry-picks:
+        # envoy 1.25.6 with patches applied:
         # https://github.com/envoyproxy/envoy/pull/25856
```
Comment (ashishb-solo) [inline bazel/repository_locations.bzl:4]: i think these comments can be removed IF we are no longer depending on envoy-fork

**Review by ben-taussig-solo — COMMENTED** [review]
Just confirmed that the filter state patch is the same as the linked PR, excluding the changelog entry and documentation changes. Looking into the dynamic metadata patch now

Update: The dynamic metadata patch looks good as well! I hit a snag trying to apply it before Jacob and Ashish clarified that it must be applied _after_ the filter state patch

File: bazel/repository_locations.bzl [hunk 4]
```diff
@@ -1,15 +1,18 @@
...
         # https://github.com/envoyproxy/envoy/pull/26311
         #   ^ add dynamic metadata to MatchingData
-        #     https://github.com/solo-io/envoy-fork/commit/71359920812f13130e657c25913c8b174852daf4
-        #     https://github.com/ashishb-solo/envoy/tree/ashishb-solo/add-dynamic-metadata-to-matchingdata
-        commit = "fa3b44a537def7ab966e2192e2693be9bede0446",
-        remote = "https://github.com/solo-io/envoy-fork",
+        #     https://github.com/envoyproxy/envoy/commit/1b66c3690422fb0d423411ebe70c777b9afbf2da
+        commit = "f89e0efddb8e4f8bad556608aa0e68fd5eae8d37",
+        remote = "https://github.com/envoyproxy/envoy",
+        patches = [
```
Comment (ashishb-solo) [inline bazel/repository_locations.bzl:12]:
```suggestion
        # the following 2 patches are needed to support the deprecated cipher passthrough and only need to be backported onto envoy v1.25.x
        patches = [
```

Comment (jbohanon) [reply bazel/repository_locations.bzl:6]: [218a00a](https://github.com/solo-io/envoy-gloo/pull/235/commits/218a00a69d2a50773a14840705cc81201fe8baf5) wdyt?

File: bazel/repository_locations.bzl (hunk 3 above)
Comment (jbohanon) [reply bazel/repository_locations.bzl]: [218a00a](https://github.com/solo-io/envoy-gloo/pull/235/commits/218a00a69d2a50773a14840705cc81201fe8baf5)

File: bazel/repository_locations.bzl (hunk 4 above)
Comment (ashishb-solo) [reply bazel/repository_locations.bzl:6]: 👍

**Review by ashishb-solo — APPROVED** [review]
💯

**Pull request merged**